src/Microsoft.EntityFrameworkCore.AutoHistory/Extensions/DbContextExtensions.cs

```diff
@@ -17,24 +19,37 @@ public static class DbContextExtensions
         /// Ensures the automatic history.
         /// </summary>
         /// <param name="context">The context.</param>
-        public static void EnsureAutoHistory(this DbContext context)
+        /// <param name="creator">User's name or email</param>
+        /// <param name="ipAddress">User's IP address</param>
+        public static void EnsureAutoHistory(this DbContext context, string creator = null, string ipAddress = null)
         {
             // Must ToArray() here for excluding the AutoHistory model.
             // Currently, only support Modified and Deleted entity.
             var entries = context.ChangeTracker.Entries().Where(e => e.State == EntityState.Modified || e.State == EntityState.Deleted).ToArray();
             foreach (var entry in entries)
             {
-                context.Add(entry.AutoHistory());
+                context.Add(entry.AutoHistory(creator, ipAddress));
             }
         }
 
-        internal static AutoHistory AutoHistory(this EntityEntry entry)
+        internal static AutoHistory AutoHistory(this EntityEntry entry, string creator = null, string ipAddress = null)
```

Comment:
I know what you want, but this will increase the usage complexity and difficulty, let me think whether have any better solution, so pending this PR.

Reply:
make AutoHistory Can inherit, and add AutoHistoryFactory.
So, if want to custom AutoHistory, just new class inherit from AutoHistory and set a custom AutoHistoryFactory.

Reply:
Yes, this is a good idea.